<?xml version="1.0" encoding="UTF-8"?>
<robot generator="Robot 6.0 (Python 3.9.6 on darwin)" generated="20250415 10:27:42.977" rpa="false" schemaversion="3">
<suite id="s1" name="Api" source="/Users/angelashi/Documents/GitHub/robotcache/api.robot">
<test id="s1-t1" name="Verify GET Request" line="7">
<kw name="Create Session" library="RequestsLibrary">
<arg>weather_api</arg>
<arg>https://www.metaweather.com</arg>
<arg>verify=true</arg>
<doc>Create Session: create a HTTP session to a server</doc>
<msg timestamp="20250415 10:27:43.127" level="INFO">Creating Session using : alias=weather_api, url=https://www.metaweather.com, headers={},                     cookies={}, auth=None, timeout=None, proxies=None, verify=True,                     debug=0 </msg>
<status status="PASS" starttime="20250415 10:27:43.127" endtime="20250415 10:27:43.130"/>
</kw>
<kw name="GET On Session" library="RequestsLibrary">
<var>${response}</var>
<arg>weather_api</arg>
<arg>/api/location/search/</arg>
<arg>params=query=taiwan</arg>
<doc>Sends a GET request on a previously created HTTP Session.</doc>
<msg timestamp="20250415 10:27:43.199" level="WARN">Retrying (RetryAdapter(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
<msg timestamp="20250415 10:27:43.437" level="WARN">Retrying (RetryAdapter(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
<msg timestamp="20250415 10:27:43.888" level="WARN">Retrying (RetryAdapter(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
<msg timestamp="20250415 10:27:43.917" level="FAIL">SSLError: HTTPSConnectionPool(host='www.metaweather.com', port=443): Max retries exceeded with url: /api/location/search/?query=taiwan (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)')))</msg>
<status status="FAIL" starttime="20250415 10:27:43.130" endtime="20250415 10:27:43.941"/>
</kw>
<kw name="Status Should Be" library="RequestsLibrary">
<arg>200</arg>
<arg>${response}</arg>
<doc>Fails if response status code is different than the expected.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.941" endtime="20250415 10:27:43.942"/>
</kw>
<kw name="Get Value From Json" library="JSONLibrary">
<var>${location_type}</var>
<arg>${response.json()}</arg>
<arg>$[0].location_type</arg>
<doc>Get Value From JSON using JSONPath</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.942" endtime="20250415 10:27:43.942"/>
</kw>
<kw name="Should Contain" library="BuiltIn">
<arg>${location_type[0]}</arg>
<arg>City</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.943" endtime="20250415 10:27:43.943"/>
</kw>
<kw name="Create Dictionary" library="BuiltIn">
<var>${headers}</var>
<arg>Authorization=Bearer ${token}</arg>
<doc>Creates and returns a dictionary based on the given ``items``.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.943" endtime="20250415 10:27:43.944"/>
</kw>
<kw name="GET On Session" library="RequestsLibrary">
<var>${response}</var>
<arg>mysession</arg>
<arg>/secure-endpoint</arg>
<arg>headers=${headers}</arg>
<doc>Sends a GET request on a previously created HTTP Session.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.944" endtime="20250415 10:27:43.945"/>
</kw>
<kw name="Create Dictionary" library="BuiltIn">
<var>${data}</var>
<arg>name=New Name</arg>
<arg>email=new@example.com</arg>
<doc>Creates and returns a dictionary based on the given ``items``.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.945" endtime="20250415 10:27:43.946"/>
</kw>
<kw name="PUT On Session" library="RequestsLibrary">
<var>${response}</var>
<arg>mysession</arg>
<arg>/users/123</arg>
<arg>json=${data}</arg>
<doc>Sends a PUT request on a previously created HTTP Session.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.946" endtime="20250415 10:27:43.947"/>
</kw>
<kw name="DELETE On Session" library="RequestsLibrary">
<var>${response}</var>
<arg>mysession</arg>
<arg>/users/123</arg>
<doc>Sends a DELETE request on a previously created HTTP Session.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.947" endtime="20250415 10:27:43.948"/>
</kw>
<kw name="Status Should Be" library="RequestsLibrary">
<arg>204</arg>
<arg>${response}</arg>
<doc>Fails if response status code is different than the expected.</doc>
<status status="NOT RUN" starttime="20250415 10:27:43.948" endtime="20250415 10:27:43.948"/>
</kw>
<status status="FAIL" starttime="20250415 10:27:43.126" endtime="20250415 10:27:43.949">SSLError: HTTPSConnectionPool(host='www.metaweather.com', port=443): Max retries exceeded with url: /api/location/search/?query=taiwan (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)')))</status>
</test>
<status status="FAIL" starttime="20250415 10:27:42.980" endtime="20250415 10:27:43.950"/>
</suite>
<statistics>
<total>
<stat pass="0" fail="1" skip="0">All Tests</stat>
</total>
<tag>
</tag>
<suite>
<stat pass="0" fail="1" skip="0" id="s1" name="Api">Api</stat>
</suite>
</statistics>
<errors>
<msg timestamp="20250415 10:27:43.199" level="WARN">Retrying (RetryAdapter(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
<msg timestamp="20250415 10:27:43.437" level="WARN">Retrying (RetryAdapter(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
<msg timestamp="20250415 10:27:43.888" level="WARN">Retrying (RetryAdapter(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /api/location/search/?query=taiwan</msg>
</errors>
</robot>
